shell/components/CommunityLinks.vue

```diff
@@ -80,10 +81,10 @@ export default {
   },
   methods: {
     show() {
-      this.$modal.show('wechat-modal');
+      this.showWeChatModal = true;
```

Comment:
> I am not so sure it's that convenient to change from a global variable to a component one.
>
>    There's the risk to end up with multiple concurrent occurrences with isolated logic
>    We are replacing a lot of code which needs to be tested in each single view
>
> Could we maybe replace the global variable to a store value?

Yeah, I think that we can do this, but first some rationale for the current approach 😄. I initially opted for the current approach because I found it to be more simple for the following reasons:

1. `AppModal.vue` doesn't need to track anything related to its own state, it's either rendered or it isn't
2. We're not effectively taking advantage of global state/events; most of the modified components represented in this PR have a direct parent => child relationship between defining a modal and controlling whether or not the modal is rendered. For example, `shell/components/AssignTo.vue` directly defines the `assignTo` modal and directly controls its visibility. No other component toggles the `assignTo` modal, so we're not really benefiting from this abstraction
3. Global state & modals are handled by `PromptModal.vue` in Dashboard. Point 2 also applies to `PromptModal`. Historically, the preferred method for managing global state for modals has been that dispatching `promptModal`. 

>  The QA costs are high, which we decided to avoid due lack of automations for each case. With a global one we should ensure the same logic everywhere.

Perhaps.. I think that the QA effort unfortunately remains the same as we are replacing the core modal implementation with something else, so we need ensure that existing modals don't regress in other ways than checking if they render or not. Maybe we would benefit from writing more e2e tests around these areas where we can.

> Also how do we manage to open the dialog using just the variable?

This is at the core of how `vue-portal` works. Rendering content in a portal will pipe its content to its portal target. `AppModal.vue` contains a portal target at its root, so conditionally rendering the component toggles the modal.